segmentation/include/pcl/segmentation/cpc_segmentation.h

```diff
@@ -0,0 +1,285 @@
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#ifndef PCL_SEGMENTATION_CPC_SEGMENTATION_H_
+#define PCL_SEGMENTATION_CPC_SEGMENTATION_H_
+
+// common includes
+#include <pcl/pcl_base.h>
+#include <pcl/point_types.h>
+#include <pcl/point_cloud.h>
+
+// segmentation includes
+#include <pcl/segmentation/supervoxel_clustering.h>
+#include <pcl/segmentation/lccp_segmentation.h>
+#include <pcl/segmentation/extract_clusters.h>
+
+// sample consensus includes
+#include <pcl/sample_consensus/sac.h>
+#include <pcl/sample_consensus/sac_model.h>
+
+#define PCL_INSTANTIATE_CPCSegmentation(T) template class PCL_EXPORTS pcl::CPCSegmentation<T>;
+#define PCL_INSTANTIATE_WeightedRandomSampleConsensus(T) template class PCL_EXPORTS pcl::WeightedRandomSampleConsensus<T>;
+
+namespace pcl
+{  
+  /** \brief A segmentation algorithm partitioning a supervoxel graph. It uses planar cuts induced by local concavities for the recursive segmentation. Cuts are estimated using locally constrained directed RANSAC.
+   *  \note If you use this in a scientific work please cite the following paper:
+   *  M. Schoeler, J. Papon, F. Woergoetter
+   *  Constrained Planar Cuts - Object Partitioning for Point Clouds
+   *  In Proceedings of the IEEE Conference on Computer Vision and Pattern Recognition (CVPR) 2015
+   *  \author Markus Schoeler (mschoeler@web.de)
+   *  \ingroup segmentation
+   */
+  template <typename PointT>
+  class CPCSegmentation : public LCCPSegmentation<PointT>
+  {
+      typedef pcl::PointXYZINormal WeightSACPointType;
+      typedef LCCPSegmentation<PointT> LCCP;
+      // LCCP typedefs
+      using typename LCCP::EdgeID;
+      using typename LCCP::EdgeIterator;
+      // LCCP methods
+      using LCCP::prepareSegmentation;
+      using LCCP::calculateConvexConnections;
+      using LCCP::applyKconvexity;
+      using LCCP::doGrouping;
+      // LCCP variables
+      using LCCP::sv_adjacency_list_;
+      using LCCP::k_factor_;
+      using LCCP::grouping_data_valid_;
+      using LCCP::sv_label_to_seg_label_map_;
+      using LCCP::sv_label_to_supervoxel_map_;
+      using LCCP::concavity_tolerance_threshold_;
+      using LCCP::seed_resolution_;
+
+    public:
+      CPCSegmentation ();
+      virtual
+      ~CPCSegmentation ();
+
+      /** \brief Merge supervoxels using local convexity. The input parameters are generated by using the SupervoxelClustering class. To retrieve the output use the relabelCloud method (inherited from LCCPSegmentation).
+       *  \param[in] supervoxel_clusters_arg Map of < supervoxel labels, supervoxels >
+       *  \param[in] label_adjacency_arg The graph defining the supervoxel adjacency relations  */
+      void
+      segment (std::map<uint32_t, typename pcl::Supervoxel<PointT>::Ptr> &supervoxel_clusters_arg,
```

Comment:
If these are "in" parameters as the docs suggest, then they can be made `const`.